common/src/test/java/io/netty/util/internal/logging/Log4J2LoggerTest.java

```diff
@@ -15,211 +15,85 @@
-
-        verify(mock).getName();
-        verify(mock).info("a");
-    }
-
-    @Test
-    public void testInfoWithException() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.message.Message;
+import org.junit.Test;
 
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.info("a", e);
+import io.netty.util.internal.ReflectionUtil;
 
-        verify(mock).getName();
-        verify(mock).info("a", e);
+/**
+ * {@linkplain Log4J2Logger} extends {@linkplain ExtendedLoggerWrapper} implements {@linkplain InternalLogger}.<br>
+ * {@linkplain ExtendedLoggerWrapper} is Log4j2 wrapper class to support wrapped loggers,
+ * so There is no need to test it's method.<br>
+ * We only need to test the netty's {@linkplain InternalLogger} interface method.<br>
+ * It's meaning that we only need to test the Override method in the {@linkplain Log4J2Logger}.
+ */
+public class Log4J2LoggerTest extends AbstractInternalLoggerTest<Logger> {
+
+    {
+        mockLog = LogManager.getLogger(loggerName);
+        logger = new Log4J2Logger(mockLog) {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public void logMessage(String fqcn, Level level, Marker marker, Message message, Throwable t) {
+                result.put("level", level.name());
+                result.put("t", t);
+                super.logMessage(fqcn, level, marker, message, t);
+            };
+        };
     }
 
     @Test
-    public void testWarn() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
-
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.warn("a");
-
-        verify(mock).getName();
-        verify(mock).warn("a");
+    public void testEXCEPTION_MESSAGE() {
+        assertEquals(getFieldValue(AbstractInternalLogger.class, "EXCEPTION_MESSAGE"),
+                getFieldValue(Log4J2Logger.class, "EXCEPTION_MESSAGE"));
     }
 
-    @Test
-    public void testWarnWithException() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
-
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.warn("a", e);
-
-        verify(mock).getName();
-        verify(mock).warn("a", e);
+    @SuppressWarnings("unchecked")
+    private static <T> T getFieldValue(Class<?> clazz, String fieldName) {
+        try {
+            Field field = clazz.getDeclaredField(fieldName);
+            if (!field.isAccessible()) {
+                assertNull(ReflectionUtil.trySetAccessible(field));
+            }
+            return (T) field.get(AbstractInternalLogger.class);
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(e);
```

Comment:
IllegalStateException

Reply:
what happend?  I runing the test in jdk8 and jdk9 but all test ok!

if there throw any RuntimeExcception, the test case must be fail.

Reply:
RuntimeExcception is a generic exception, in this use case it makes more sense to throw an IllegalStateException instead.

Reply:
@johnou Thanks, you solved my doubts. 
I just solved the problem from the function, not from the use of scenes.